base-system/usrroot/usr/lib/hsync/libhapply.so

```diff
@@ -404,3 +406,23 @@ set_wallpaper() {
 	log "-Couldn't find any background to restore"
 	return 1 # error
 }
+
+set_scheduled_shutdown() {
+	local SHUTDOWN_TIME SHUTDOWN_TIME_EPOCH CURRENT_TIME_EPOCH SECONDS_TO_SHUTDOWN MINUTES_TO_SHUTDOWN
+	SHUTDOWN_TIME="$(get_directives_var Global ShutdownTime)"
+	log "Cancelling scheduled shutdown (if any)"
+	shutdown -c
+	if [ "$SHUTDOWN_TIME" = "none" ]; then
+		return 0
+	fi
+	SHUTDOWN_TIME_EPOCH=$(date -d "$SHUTDOWN_TIME" +%s)
+	CURRENT_TIME_EPOCH=$(date +%s)
+	SECONDS_TO_SHUTDOWN=$((SHUTDOWN_TIME_EPOCH-CURRENT_TIME_EPOCH))
+	if [ "$SECONDS_TO_SHUTDOWN" -lt 0 ]; then
+		log "Current shutdown is for a time that already happened, skiping scheduled shutdown."
+		exit 0
+	fi
+	MINUTES_TO_SHUTDOWN=$((SECONDS_TO_SHUTDOWN/60))
+	log "Scheduling shutdown in $MINUTES_TO_SHUTDOWN minute(s)"
+	shutdown +"$MINUTES_TO_SHUTDOWN"
```

Comment:
Also, let's consider the option to use [*systemd-shutdownd*](https://www.freedesktop.org/software/systemd/man/systemd-poweroff.service.html). It might have some advantages over *shutdown*.

Reply:
On this side fater some searching seems like `shutdown` is a link to `systemd-shutdown`, so they do the same thing. Unless there's something I'm not seeing, feel free to mention if that's the case